app/lib/backend_api_logic/proxy_extension.rb

```diff
@@ -20,6 +20,7 @@ module ProxyExtension
     protected
 
     def validate_backend_api?
+      return unless account # TODO: remove this line when we remove the Rolling Update :)
```

Comment:
Modifying the models and logic only for tests to pass is calling for huge problems.
Tests are not agnostics, they precisely need to know what are the objects they are testing.

You can make a test more generic but if you can't then ... too bad you can't